### ISSUE TITLE
NPM

### ISSUE THREAD

**Issue by pyrsmk** [cferdinandi/smooth-scroll#241]
It seems the package is not available on NPM anymore

**Comment by lkraav** [comment]
Is it really needed? `npm install cferdinandi/smooth-scroll` will save the dep as `github:cferdinandi/smooth-scroll`.

**Comment by pyrsmk** [comment]
Ah that's right, I forgot the namespace ^^
But it's easier to have it directly on NPM ;)

**Comment by pyrsmk** [comment]
On the other hand, I don't think my installation process (because it's integrated in a framework I'm using for my enterprise) will install it as expected. It's a big issue to me.

**Comment by pyrsmk** [comment]
Never mind, it worked very well. Sorry and thanks for your help ;)

**Comment by cferdinandi** [comment]
It actually was never available on NPM. I manage a few dozen OSS projects, so for the sake of sanity, I support package managers exclusively through Git hooks. Glad you got things figured out, and @lkraav, thanks for jumping in!

**Comment by mockdeep** [comment]
This has been a pain when running `npm shrinkwrap`. It would be nice to be able to pin at a specific version via our `package.json`. It's not that hard to push to npm when you update the version...

**Comment by cferdinandi** [comment]
@mockdeep Totally get it. You actually already can pin a specific version directly from GitHub:

```javascript
git://github.com/user/repo.git#v0.0.1
```

**sources: **

- http://www.devthought.com/2012/02/17/npm-tricks/
- http://stackoverflow.com/a/16350674
- https://docs.npmjs.com/files/package.json#git-urls-as-dependencies

I get asked about this a lot (along with Bower and other popular package managers). Here's the thing: I actively maintain [a few dozen open source projects](http://gomakethings.com/free-stuff/). Every package manager, even the popular ones like NPM, introduce additional maintenance and overhead in my workflow on something that I spend a lot of time developing and give away for free.

The popular ones, including NPM, already support all of the important things directly from GitHub without me needing to register and host code there, so there's extra work on my end (across dozens of projects) for a minimal increase in value for the end users.

In other words, I hear you, but it's just not gonna happen.

**Comment by pyrsmk** [comment]
Legit ^^

**Comment by mockdeep** [comment]
We decided to run with [zenscroll](https://github.com/zengabor/zenscroll) instead. We were getting errors when trying to shrinkwrap this like:

```
npm ERR! invalid: have smooth-scroll@8.0.0 (expected: github:cferdinandi/smooth-scroll)
```

It's hard enough getting npm to behave consistently without throwing stuff like this in the mix, and definitely not worth the struggle for something as small as a smooth scrolling library.

**Comment by cferdinandi** [comment]
:+1:
